src/main/java/com/tsunderebug/speedrun4j/user/User.java

```diff
@@ -27,30 +33,123 @@ public class User {
 	private Resource twitter;
 	private Resource speedrunslive;
 	private Link[] links;
+	private UserAssets assets;
+
+	/**
+	 * Get user from ID or Username
+	 * @param id ID or username
+	 * @return speedrun.com User
+	 */
+	public static User fromID(String id) {
 
-	public static User fromID(String id) throws IOException {
 		Gson g = new Gson();
-		URL u = new URL(Speedrun4J.API_ROOT + "users/" + id);
-		HttpURLConnection c = (HttpURLConnection) u.openConnection();
+		URL u;
+
+		try {
+			u = new URL(Speedrun4J.API_ROOT + "users/" + id);
+		} catch (MalformedURLException e) {
+			throw new SpeedrunUrlException("Could not get URL from ID.");
+		}
+
+		HttpURLConnection c;
+
+		try {
+			c = (HttpURLConnection) u.openConnection();
+		} catch (IOException e) {
+			throw new SpeedrunUrlException("Could not connect to URL from ID.");
+		}
+
 		c.addRequestProperty("User-Agent", Speedrun4J.USER_AGENT);
-		InputStreamReader r = new InputStreamReader(c.getInputStream());
+		InputStreamReader r;
+
+		try {
+			r = new InputStreamReader(c.getInputStream());
+		} catch (IOException e) {
+			throw new SpeedrunJsonException("Could not read input from input stream.");
+		}
+
 		UserData d = g.fromJson(r, UserData.class);
-		r.close();
+
+		try {
+			r.close();
+		} catch (IOException e) {
+			throw new SpeedrunJsonException("Could not close input stream.");
+		}
+
 		return d.data;
+
 	}
 
-	public static User fromApiKey(String key) throws IOException {
+	/**
+	 * Get user from API Key
+	 * @see <a href="https://github.com/speedruncomorg/api/blob/master/authentication.md">Authentication Documentation</a>
+	 * @param key API key from speedrun.com
+	 * @return User from API Key
+	 */
+	public static User fromApiKey(String key) {
+
 		Gson g = new Gson();
-		URL u = new URL(Speedrun4J.API_ROOT + "profile");
-		HttpURLConnection c = (HttpURLConnection) u.openConnection();
+		URL u;
+
+		try {
+			u = new URL(Speedrun4J.API_ROOT + "profile");
+		} catch (MalformedURLException e) {
+			throw new SpeedrunUrlException("Could not get URL from API Key.");
```

Comment:
Same as above with regards to chaining and including useful data in the exception message.